src/passes/literalExpressionEvaluator/literalExpressionEvaluator.ts

```diff
@@ -128,15 +129,43 @@ function evaluateBinaryLiteral(node: BinaryOperation): RationalLiteral | boolean
   if (left === null || right === null) {
     // In some cases a binary expression could be calculated at
     // compile time, even when only one argument is a literal.
+
+    let rightExpression;
+    if (left === null) {
+      rightExpression = true;
+    } else {
+      rightExpression = false;
+    }
+
     const notNullMember = left ?? right;
     if (notNullMember === null) {
       return null;
     } else if (typeof notNullMember === 'boolean') {
       switch (node.operator) {
         case '&&': // false && x = false
-          return notNullMember ? null : false;
+          if (notNullMember === true) {
+            if (rightExpression) {
+              replaceNode(node, node.vRightExpression);
+            } else {
+              replaceNode(node, node.vLeftExpression);
+            }
+            return null;
+          } else {
+            return false;
+          }
+        //return notNullMember ? null : false;
         case '||': // true || x = true
-          return notNullMember ? true : false;
+          if (notNullMember === false) {
+            if (rightExpression) {
+              replaceNode(node, node.vRightExpression);
+            } else {
+              replaceNode(node, node.vLeftExpression);
+            }
+            return false;
```

Comment:
This part should return `null` instead of `false`. This is probably a mistake that was left in the last pr that was made related to this pass. Obviously this behavior is not being tested which is not good. If you have a simple contract like: 
```
// SPDX-License-Identifier: AGPL-3.0-only
pragma solidity ^0.8.14;

contract WARP {
  function f(bool a) public pure returns (bool) {
    return false || a;
  }
}
```
Calling function `f(true)` should return `false`, which is not correct. Please fix this and add a test case as well.